packages/medusa/src/utils/__tests__/transaction/transaction-builder.ts

```diff
@@ -0,0 +1,376 @@
+
+      expect(transactionBuilder.build()).toEqual({})
+    })
+
+    test("deleteAction - nested", () => {
+      transactionBuilder
+        .addAction("foo", {})
+        .addAction("bar", {})
+        .addAction("axe", {})
+        .deleteAction("bar")
+
+      expect(transactionBuilder.build()).toEqual({
+        action: "foo",
+        next: {
+          action: "axe",
+        },
+      })
+    })
+
+    test("pruneAction", () => {
+      transactionBuilder
+        .addAction("foo", {})
+        .addAction("bar", {})
+        .addAction("axe", {})
+        .addAction("step", {})
+        .pruneAction("bar")
+      expect(transactionBuilder.build()).toEqual({
+        action: "foo",
+      })
+    })
+
+    test("build", () => {
+      transactionBuilder.addAction("foo", {})
+      expect(transactionBuilder.build()).toEqual({
+        action: "foo",
+      })
+    })
+  })
+
+  describe("Composing Complex Transactions", () => {
+    const loadedFlow = {
+      next: {
+        action: "createProduct",
+        saveResponse: true,
+        next: {
+          action: "attachToSalesChannel",
+          saveResponse: true,
+          next: {
+            action: "createPrices",
+            saveResponse: true,
+            next: {
+              action: "createInventoryItems",
+              saveResponse: true,
+              next: {
+                action: "attachInventoryItems",
+                noCompensation: true,
+              },
+            },
+          },
+        },
+      },
+    }
+
+    it("should load a transaction and add two steps", () => {
+      const builder = new TransactionBuilder(loadedFlow)
+      builder
+        .addAction("step_1", {}, { saveResponse: true })
+        .addAction("step_2", {}, { saveResponse: true })
+
+      expect(builder.build()).toEqual({
+        action: "createProduct",
+        saveResponse: true,
+        next: {
+          action: "attachToSalesChannel",
+          saveResponse: true,
+          next: {
+            action: "createPrices",
+            saveResponse: true,
+            next: {
+              action: "createInventoryItems",
+              saveResponse: true,
+              next: {
+                action: "attachInventoryItems",
+                noCompensation: true,
+                next: {
+                  action: "step_1",
+                  saveResponse: true,
+                  next: {
+                    action: "step_2",
+                    saveResponse: true,
+                  },
+                },
+              },
+            },
+          },
+        },
+      })
+    })
+
+    it("should load a transaction, add 2 steps and move step_1 to run in parallel with createProduct", () => {
+      const builder = new TransactionBuilder(loadedFlow)
+      builder
+        .addAction("step_1", {}, { saveResponse: true })
+        .addAction("step_2", {}, { saveResponse: true })
+        .moveAction("step_1", "createProduct", true)
```

Comment:
question: From a previous tests, I thought the moveAction would have moved step_1 before createProduct and that to make them parallel we should use merge. This might be due to the true value at the last argument, maybe having an object to make it explicit could be nice

Reply:
Well, I think for sake of simplicity I can only keep the `merge` to do that, which internally is just calling `move` with this parameter to run in parallel.

Reply:
That would be great, one concept to do the job which will prevent some questions 👍